### PR TITLE
Fix losses

### PR DESCRIPTION
**Pull request by SphericalPotatoInVacuum**

### CODE REVIEW

File: src/config/baseline.py [hunk 1]
```diff
@@ -3,31 +3,29 @@
...
+        style_dir = Path('data/IMGUR5K')
+        download_dataset('IMGUR5K')
         batch_size = 4
-        train_dataloader = DataLoader(BaselineDataset(style_dir / 'train'), shuffle=True, batch_size = batch_size)
-        val_dataloader = DataLoader(BaselineDataset(style_dir / 'val'), batch_size = batch_size)
+        train_dataloader = DataLoader(BaselineDataset(style_dir / 'train'), shuffle=True, batch_size=batch_size)
+        val_dataloader = DataLoader(BaselineDataset(style_dir / 'val'), batch_size=batch_size)
 
-        total_epochs = 1 #20
+        total_epochs = 1  # 20
```
Comment (grenlayk) [inline src/config/baseline.py:28]: I am sure we should either delete comment or set `total_epochs = 20`

File: src/training/baseline.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import torch
 from src.logger.simple import Logger
 from src.storage.simple import Storage
-from src.losses import ocr
+from src.losses.STRFL import OCRLoss
```
Comment (grenlayk) [inline src/training/baseline.py:4]: Not sure we need to change it here -- in our paper we described that in first model we used old OCR loss.

File: src/training/baseline.py [hunk 3]
```diff
@@ -29,19 +31,17 @@ def __init__(self,
         self.total_epochs = total_epochs
         self.logger = logger
         self.storage = storage
-        self.ocr_loss = ocr.OCRLoss().to(device)
+        self.ocr_loss: nn.Module = OCRLoss().to(device)
```
Comment (grenlayk) [inline src/training/baseline.py:34]: Well... If you decided to restrict this piece of code too, it might be better to pass loss function from config and mark variable type in function's args.

File: src/training/stylegan.py [hunk 4]
```diff
@@ -41,39 +41,44 @@ def __init__(self,
...
-        self.coef_ocr = ocr_coef
-        self.coef_cycle = coef_cycle_loss
-        self.coef_recon = coef_reconstruction_loss
+        self.typeface_loss = typeface_loss.to(device)
+        self.ocr_loss = ocr_loss.to(device)
+        self.perc_loss = perc_loss.to(device)
         self.style_embedder = style_embedder
         self.content_embedder = content_embedder
-        self.l1 = torch.nn.L1Loss() 
+        self.l1 = torch.nn.L1Loss()
```
Comment (grenlayk) [inline src/training/stylegan.py:55]: Don't we need to put in onto device? Also I am not sure that we should initialise it here and not in config.

File: src/training/stylegan.py [hunk 5]
```diff
@@ -41,39 +41,44 @@ def __init__(self,
...
+
             style_embeds = self.style_embedder(style_batch)
             content_embeds = self.content_embedder(content_batch)
-            self.optimizer.zero_grad()
 
             res = self.model(content_embeds, style_embeds)
-            ocr_loss, recognized = self.ocr_loss(res, label_batch, return_recognized=True)
+            ocr_loss, recognized = self.ocr_loss(res, content_label_batch, return_recognized=True)
             words = []
             for word in recognized:
```
Comment (grenlayk) [inline src/training/stylegan.py:83]: I was thinking about this part of code where we collect recognised words and maybe we should make it a function inside of OCR (but I am not sure).

**Review by grenlayk — COMMENTED** [review]
lgtm except some minor stuff

File: src/training/stylegan.py (hunk 5 above)
Comment (SphericalPotatoInVacuum) [reply src/training/stylegan.py:85]: We could move drawing the words inside util/

File: src/training/stylegan.py [hunk 6]
```diff
@@ -41,77 +45,77 @@ def __init__(self,
...
         self.model.train()
         self.content_embedder.train()
         self.style_embedder.train()
 
-        for style_batch, content_batch, label_batch, style_label_batch in self.train_dataloader:
-            if max(len(label) for label in label_batch) > 25:
+        # style_batch - images containing the style we want to imitate
+        # conent_batch - rendered images containing the conent we want to draw
+        # label_batch - text labels of content batch
+        # style_label_batch - text labels of style batch
```
Comment (nikoryagin) [inline src/training/stylegan.py:70]: style_label_batch is rendered pictures of content presented in style batch
probably we should change the name, but I can’t come up with the better one

**Comment by nikoryagin** [comment]
lgtm except for tiny little naming thing

File: src/training/stylegan.py (hunk 6 above)
Comment (SphericalPotatoInVacuum) [reply src/training/stylegan.py:70]: `unstylized_style_batch`?
